java/mykola-melnyk/src/com/zoolatech/lecture2/tasks/_2/SimpleRobot.java

```diff
@@ -0,0 +1,226 @@
+                x = scanner.nextInt();
+            } while (x < 0 || x >= W);
+
+            do {
+                System.out.println("Enter y (height) axis coordinates " +
+                        "(must be > 0 and < room height):");
+                y = scanner.nextInt();
+            } while (y < 0 || y >= H);
+        }
+
+
+        System.out.println("Default direction is South. Set custom? (y/n)");
+        if (scanner.next().charAt(0) == 'y') {
+            System.out.println("Possible variants are: N, E, S, W");
+            if (scanner.next().charAt(0) == 'N') {
+                dir = Direction.NORTH;
+            } else if (scanner.next().charAt(0) == 'E') {
+                dir = Direction.EAST;
+            } else if (scanner.next().charAt(0) == 'S') {
+                dir = Direction.SOUTH;
+            } else if (scanner.next().charAt(0) == 'W') {
+                dir = Direction.WEST;
+            } else {
+                System.out.println("Unknown parameter. Setting to default (East).");
+            }
+        } else dir = Direction.EAST;  //set default dir to South
+
+        if (dir == Direction.WEST) {  //convert enum value to human-readable output
+            compass = "West";
+        } else if (dir == Direction.EAST) {
+            compass = "East";
+        } else if (dir == Direction.NORTH) {
+            compass = "North";
+        } else if (dir == Direction.SOUTH) {
+            compass = "South";
+        }
+    }
+
+    public void move(int steps) {
+
+        System.out.println("Moving " + steps + " step(s) to " + compass + ".");
+        if (steps > 0) {
+
+            if (dir == Direction.EAST) {
+                if ((x + steps) < W) {
+                    x += steps;
+                } else {
+                    x = W - 1;
+                    System.out.println("Reached Eastern wall! Change direction");
+                }
+            } else if (dir == Direction.WEST) {
+                if ((x - steps) >= 0) {
+                    x -= steps;
+                } else {
+                    x = 0;
+                    System.out.println("Reached Western wall! Change direction");
+                }
+            } else if (dir == Direction.SOUTH) {
+                if ((y + steps) < H) {
+                    y += steps;
+                } else {
+                    y = H - 1;
+                    System.out.println("Reached Southern wall! Change direction");
+                }
+            } else if (dir == Direction.NORTH) {
+                if ((y - steps) >= 0) {
+                    y -= steps;
+                } else {
+                    y = 0;
+                    System.out.println("Reached Northern wall! Change direction");
+                }
+            }
+        } else {
+            System.out.println("Robot made no moves!");
+        }
+    }
+
+    public void moveAndShow(int steps) {
+
+        System.out.println("Moving " + steps + " step(s) to " + compass + ".");
+        if (steps > 0) {
```

Comment:
I think we can remove this if-else block and just call two methods: `move` and `showPosition`

Reply:
Resolved: rewrote the program code completely.